phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -1938,6 +1946,44 @@ private void populateFamilyPropsList(Map<String, PName> familyNames, Map<String,
         }
     }
 
+    /**
+     * Get TTL defined for Index or View in its parent hierarchy as defining TTL directly on index or view is
+     * not allowed. View on SYSTEM table is not allowed and already handled during plan compilation.
+     * @param parent
+     * @return appropriate TTL for the entity calling the function.
+     * @throws TableNotFoundException
+     */
+    private Long getTTLFromParent(PTable parent) throws TableNotFoundException {
+        return (parent.getType() == TABLE) ? Long.valueOf(parent.getPhoenixTTL()) :
+                (parent.getType() == VIEW ? getTTLFromAncestor(parent) : null);
+    }
+
+    /**
+     * Get TTL defined for the given View according to its hierarchy.
+     * @param view
+     * @return appropriate TTL from Views defined above for the entity calling.
+     * @throws TableNotFoundException
+     */
+    private Long getTTLFromAncestor(PTable view) throws TableNotFoundException {
+        try {
+            return view.getPhoenixTTL() != PHOENIX_TTL_NOT_DEFINED ? Long.valueOf(view.getPhoenixTTL()) :
+                    (checkIfParentIsTable(view) ? connection.getTable(new PTableKey(null ,
+                            view.getPhysicalNames().get(0).getString())).getPhoenixTTL() :
+                            getTTLFromAncestor(connection.getTable(new PTableKey(connection.getTenantId(), view.getParentName().getString()))));
+        } catch (TableNotFoundException tne) {
+            return getTTLFromAncestor(connection.getTable(new PTableKey(null, view.getParentName().getString())));
+        }
+    }
+
+    private boolean checkIfParentIsTable(PTable view) {
+        PName parentName = view.getParentName();
+        if (parentName == null) {
+            //means this is a view on dataTable
+            return true;
+        }
+        return parentName.getString().equals(view.getPhysicalName().getString());
```

Comment:
are we checking that parent name is same as view name?

Reply:
No here we are checking if if parent name is same as base table name, to check if parent is table of not!